File: EcoBookAiBackend/src/main/java/com/ecobook/dto/UserConsentStatusDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class UserConsentStatusDTO {
    private Boolean platformConsentGiven;
    private LocalDateTime platformConsentGivenAt;
    private Boolean aiConsentEnabled;
    private LocalDateTime aiConsentGivenAt;
    private LocalDateTime aiConsentRevokedAt;
    private List<ConsentRecordDTO> history;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/UserConsentStatusDTO.kt
package com.ecobook.dto

data class UserConsentStatusDTO(
    val platformConsentGiven: Boolean = false,
    val platformConsentGivenAt: String? = null,
    val aiConsentEnabled: Boolean = false,
    val aiConsentGivenAt: String? = null,
    val aiConsentRevokedAt: String? = null,
    val history: List<ConsentRecordDTO> = emptyList()
)
